frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -1053,11 +1062,12 @@ function ReaderPaging:onGotoPageRel(diff)
                 if not at_end(y) then
                     goto_end(y)
                 else
-                    goto_next_page()
+                    page_turn_requested = goto_next_page()
                 end
             end
         end
     end
+    if no_page_turn and page_turn_requested then return end -- see ReaderPaging:onGotoViewRel
```

Comment:
`goto_next_page()` would just not set things up ready for next page, and we can just return, without leftovers ?

(Not fond of the sound of `page_turn_requested` - again feels like something we would do :) I would use `would_turn_page` or something better.)

Reply:
Yes, when `would_turn_page`, nothing should be done here, and ReaderSearch continues searching forward/backward. Any leftovers are local.